internal/backends/sqlite/metadata/registry.go

```diff
@@ -244,24 +258,42 @@ func (r *Registry) CollectionCreate(ctx context.Context, dbName, collectionName
 		return false, nil
 	}
 
+	var tableName string
 	h := fnv.New32a()
 	must.NotFail(h.Write([]byte(collectionName)))
 	s := h.Sum32()
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/2760
+	for {
+		tableName = fmt.Sprintf("%s_%08x", strings.ToLower(collectionName), s)
+		if strings.HasPrefix(tableName, reservedTablePrefix) {
+			tableName = "_" + tableName
+		}
 
-	tableName := fmt.Sprintf("%s_%08x", strings.ToLower(collectionName), s)
-	if strings.HasPrefix(tableName, reservedTablePrefix) {
-		tableName = "_" + tableName
-	}
+		q := fmt.Sprintf("CREATE TABLE %[1]q (%[2]s TEXT NOT NULL CHECK(%[2]s != '')) STRICT", tableName, DefaultColumn)
+		_, err = db.ExecContext(ctx, q)
 
-	q := fmt.Sprintf("CREATE TABLE %[1]q (%[2]s TEXT NOT NULL CHECK(%[2]s != '')) STRICT", tableName, DefaultColumn)
-	if _, err = db.ExecContext(ctx, q); err != nil {
-		return false, lazyerrors.Error(err)
+		var se *sqlite3.Error
+
+		// When a table already exists, SQLite error code sqlite3lib.SQLITE_ERROR returned is too generic.
+		// So we also check that error message contains `already exists` text.
+		// The se.Error() looks like `SQL logic error: table "XXX" already exists (1)`.
+		// See also https://www.sqlite.org/rescode.html#error.
+		if errors.As(err, &se) && se.Code() == sqlite3lib.SQLITE_ERROR && strings.Contains(se.Error(), "already exists") {
```

Comment:
We should check if the table already exists in `colls` instead of trying to create it and handle the error (badly)

Reply:
That's much better. Let's do that :hugs: